.github/workflows/ci-actions-incremental.yml

```diff
@@ -1041,7 +1041,7 @@ jobs:
       - name: Setup GraalVM
         id: setup-graalvm
         uses: graalvm/setup-graalvm@v1
-        if: startsWith(matrix.os-name, 'windows')
+        if: "startsWith(matrix.os-name, 'windows') || startsWith(matrix.category, 'Misc1')"
```

Comment:
> Judging by the commit message in [c3e1d75](https://github.com/quarkusio/quarkus/commit/c3e1d75a675839754ddd244f0106b3e43d3b4eaf), in order for the integration to work one needs to have Mandrel or GraalVM locally installed which is non-optimal as the default is to use the builder-image (and saves people from a lot of trouble).
> 
> That seems OK for the time being, but IMO we should make it work with the builder image as well.

I'm not sure about the real benefit of getting the builder image `JAVA_HOME` to be used to run the surefire unit tests. It can probably be done but I have doubts about doing it:

* First, the way I describe the feature (see https://github.com/quarkusio/quarkus/issues/36822), this feature would be used by a subset of Quarkus users, those that have the need to bring a library/component that requires some native image configuration, or those that use JDK APIs that require extensive native image configuration. Since it doesn't affect all Quarkus users, I think it's a small price to pay to ask those to install GraalVM locally and run the `native-with-agent` profile with their unit tests.

* Secondly, I see this feature as something that gives you a head start with the required native image configuration. So you would run it once at the start, get the configuration files, add them to your source control and extend as needed. Maybe you would run it again once you have made substantial changes or upgraded to a newer version. The point I'm trying to make here is that you would not be running this profile very often.

So, combining the two points above: the fact that it only affects a subset of Quarkus users and this is something that you would only run once in a while, I'm not sure of its benefits. I agree that it could be useful to remove the need for a local installation, and maybe it's quite an easy thing to do, but right now I'm not convinced.

> In any case, I would prefer the native-image agent integration test to be moved to a separate IT category and only use a local installation when testing it, to keep all testing consistent except for the actual cases that need to run in a different setup.

Yeah that makes sense. I went for a quick solution here to see if the CI configuration change would work, but I agree that such integration test should be moved to its own category. I'll try to do that within this PR.